crates/sui-framework/sources/governance/sui_system.move

```diff
@@ -460,6 +483,20 @@ module sui::sui_system {
         // Validator reports are only valid for the epoch.
         // TODO: or do we want to make it persistent and validators have to explicitly change their scores?
         self.validator_report_records = vec_map::empty();
+
+        event::emit(
+            SystemEpochInfo {
+                epoch: self.epoch,
+                reference_gas_price: self.reference_gas_price,
+                total_stake: delegation_stake + validator_stake,
+                storage_fund_inflows: storage_charge + (storage_fund_reinvestment_amount as u64),
```

Comment:
are `storage_fund_balance` and `computation_charge` updated after each txn execution? are they reset around end of epochs?

Reply:
> are storage_fund_balance and computation_charge updated after each txn execution?

No they come in once every epoch when this function `advance_epoch` is called.